### PR TITLE
Major docker changes & Makefile added

### PR DESCRIPTION
**Pull request by toikarin-vincit**
- Docker changes
   - Dockerfile now uses builder for creating a virtualenv
   - Image contains /hitas/venv and hitas/backend directories
     - /hitas/venv is built by the builder
     - /hitas/backend contains only the necessary files for running the app
     - everything is owned by `hitas` user
   - Default image only now supports uwsgi. It exposes uwsgi port as 8888
   - docker-compose now starts nginx instance with port 8080 open. It
     will serve django app over uwsgi and static media files.
 - uwsgi support added (it was broken)
 - Added Makefile
   - contains a list of simple day-to-day commands
     - `make build` starts a docker-compose environment
     - `make tests` runs tests
     - `make format` runs linters
     - `make dump` takes a new initial datamodel dump

### CODE REVIEW

File: backend/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uwsgi_param CONTENT_LENGTH $content_length;
+        uwsgi_param REQUEST_URI $request_uri;
+        uwsgi_param PATH_INFO $document_uri;
+        uwsgi_param DOCUMENT_ROOT $document_root;
+        uwsgi_param SERVER_PROTOCOL $server_protocol;
+        uwsgi_param REMOTE_ADDR $remote_addr;
+        uwsgi_param REMOTE_PORT $remote_port;
+        uwsgi_param SERVER_ADDR $server_addr;
+        uwsgi_param SERVER_PORT $server_port;
+        uwsgi_param SERVER_NAME $server_name;uwsgi_param QUERY_STRING $query_string;
```
Comment (ranta) [inline backend/nginx.conf:28]: Missing newline

Comment (toikarin-vincit) [reply backend/nginx.conf]: fixed

**Review by ranta — DISMISSED** [review]
Nice, I like the added Makefile

File: backend/docker-compose.yml [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+      APPLY_MIGRATIONS: 1
+      COLLECT_STATIC: 1
+      LOAD_INITIAL_DATASET: 1
+      DEBUG: "True"
+      ALLOWED_HOSTS: "*"
+      SECRET_KEY: top-secret-development-key
+      DATABASE_URL: postgres://hitas:hitas@postgres/hitas
     ports:
-      - "8000:8000"
+      - 8888
```
[the remark this answers is not in the document]
Comment (toikarin-vincit) [reply backend/docker-compose.yml:16]: Port used:
 - 8888 uwsgi (in hitas container)
 - 8080 http (in nginx container)
 - 8000 http (when running locally)
 
 django uses 8000 by default so I opted to change the containerized port to 8080 so it's possible to run them both at the same time (as 8080 is published).
 for uwsgi it does not matter that much which is used but I used different port to differentiate between protocols

**Review by miqu — APPROVED** [review]

**Review by ranta — APPROVED** [review]

**Pull request merged**